### PR TITLE
Create Files And Association Relationships 🧑‍🤝‍🧑 

### PR DESCRIPTION
**Pull request by gisachris**
## IN This PR I present Files And Association Relationships 🧑‍🤝‍🧑 :
 
Below are some of my implementations as well as going over the project requirements 🤝 .
## Requirements
- Setup ruby workflow and linters 🔍.
- use GitFlow 🌊.
- Create a class `Classroom`🏛️ with instance variables and accessor🌗  `@label` .
- Create the `has-many`/`belongs-to` 🍯 relationship between `Classroom` 🏛️ and `Student` 👨‍🎓.
- Create a class `Book` with instance variables and accessors🌗 `@title` 🎟️ and `@author` 🍂.
- Create a class `Rental` with instance variables and accessors🌗 `@date` ,`@book`📚  and `@person`🤺. 
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental` 📜.

## Achievements
- Setup ruby workflow and linters 🔍.✅
- use GitFlow 🌊.✅
- Create a class `Classroom`🏛️ with instance variables and accessor🌗  `@label` .✅
- Create the `has-many`/`belongs-to` 🍯 relationship between `Classroom` 🏛️ and `Student` 👨‍🎓.✅
- Create a class `Book` with instance variables and accessors🌗 `@title` 🎟️ and `@author` 🍂.✅
- Create a class `Rental` with instance variables and accessors🌗 `@date` ,`@book`📚  and `@person`🤺. ✅
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental` 📜.✅

## Conclusion
**We are looking forward to hearing 👂  from you Our CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,9 +1,12 @@
 require_relative 'person'
 
 class Student < Person
+  attr_accessor :classroom
+
   def initialize(classroom, age, name = 'unknown', parent_permission: true)
-    super(age, name, parent_permission)
+    super(age, name, parent_permission: parent_permission)
     @classroom = classroom
+    classroom.add_student(self)
```
Comment (franclobo) [inline student.rb]: - [ ] Please define a method classroom that serves as the 'belongs-to' side (a student belongs to a classroom). Look at this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) with examples. The classroom method will be identical to the owner method in the example.

File: books.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (franclobo) [inline books.rb]: - [ ] Please 🙏🏽 add a method for adding rentals should be present. Link to the example in a [previous lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

File: person.rb [hunk 3]
```diff
@@ -10,6 +10,7 @@ def initialize(age, name = 'unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
```
Comment (franclobo) [inline person.rb]: - [ ] Please 🙏🏽 add a method for adding rentals should be present. Link to the example in a [previous lesson](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship). The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @gisachris, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️ 
- Correct Git flow ✔️ 
- Correct associations between classes and objects ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gisachris** [comment]
Thank you @franclobo for the great review. I will add The required implementations 👍

**Comment by gisachris** [comment]
## IN This PR We present Changes♻️ to Association Relationships🧑‍🤝‍🧑:
 
Below are some of my implementations as well as going over the project requirements 🤝 .
## Changes Required ♻️ 
- Please define a method `classroom` that serves as the `belongs-to` side `(a student belongs to a classroom)`
- add a method for adding rentals in both `people` 🧑‍🤝‍🧑 and `book` 📚 classes. 

## Achievements
- Please define a method `classroom` that serves as the `belongs-to` side `(a student belongs to a classroom)` ✅ 
- add a method for adding rentals in both `people` 🧑‍🤝‍🧑 and `book` 📚 classes. ✅ 

## Conclusion
**We are looking forward to hearing 👂  from you The CR in terms of what can be better implemented 👨‍💻  for the project thankyou!. And we look forward to hearing from you soon.**  🙇

**Review by danmainah — APPROVED** [review]
Hi @gisachris , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the changes requested to achieve all the associations.
✔️ Professional written README file with all sections required.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gisachris** [comment]
Thankyou @danmainah I am grateful 🙏

**Pull request merged**